### PR TITLE
Theme can now be changed on changing the state of component 

### PR DESCRIPTION
**Pull request by hetanthakkar1**
**What kind of change does this PR introduce?**

Added ability to change the theme for a component tree via state change
Fixed issue #2754

### CODE REVIEW

**Comment by hetanthakkar1** [comment]
Sorry for the multiple commits that resulted because of multiple commits I did for testing my changes

**Comment by hetanthakkar1** [comment]
Demo Code which can is now supported:

image



import { ThemeProvider, Button } from "react-native-elements";
import React, { useState } from "react";
const theme1 = {
  Button: {
    titleStyle: {
      color: "white",
    },
  },
};
const theme2 = {
  Button: {
    titleStyle: {
      color: "yellow",
    },
  },
};
export default function App() {
  const [theme, setTheme] = useState(theme1);
  return (
    <ThemeProvider theme={theme}>
      <Button
        title="My Button"
        onPress={() => {
          if (theme == theme1) {
            setTheme(theme2);
          } else {
            setTheme(theme1);
          }
        }}
      />
      <Button
        title="My 2nd Button"
        onPress={() => {
          if (theme == theme1) {
            setTheme(theme2);
          } else {
            setTheme(theme1);
          }
        }}
      />
    </ThemeProvider>
  );
}

**Comment by hetanthakkar1** [comment]
@flyingcircle  You can use my above code to test my changes

**Comment by hetanthakkar1** [comment]
@flyingcircle  Could you please review!?

File: src/config/ThemeProvider.tsx [hunk 1]
```diff
@@ -76,6 +81,7 @@ export default class ThemeProvider extends React.Component<
   updateTheme = (updates: RecursivePartial<FullTheme>) => {
     this.setState(({ theme }) => ({
       theme: deepmerge(theme, updates),
+      update: true,
```
Comment (flyingcircle) [inline src/config/ThemeProvider.tsx:84]: I don't understand the use of update and setting it true here. Are we saying that if you run `updateTheme` then changing the `theme` prop will stop working? How about we just go without the update variable and just update the `theme` as it is passed in.

**Review by flyingcircle — CHANGES_REQUESTED** [review]

**Comment by hetanthakkar1** [comment]
@flyingcircle  So we are updating theme at 2 places in my code .One at the updateTheme method and other at getDerivedStateFromProps method so we dont want to execute both of them to do the same.So if updateTheme function will update theme it will set the update to true and so the updation in getDerivedStateFromProps wont take place.Removing that update state will also fail 1 test case

**Comment by flyingcircle** [comment]
@hetanthakkar1 What I'm asking is if a user updated the theme using `updateTheme`and then updates it later by changing the `theme` prop, the theme wouldn't update the second time right? I don't see where in this code that `update` gets switched back to `false`.

**Comment by hetanthakkar1** [comment]
@flyingcircle  I got your point and I have made the necessary changes. Can you please review

**Comment by hetanthakkar1** [comment]
@flyingcircle Okay i got your point. I am thinking on it. Please ignore the last commit . I pushed it by mistake that it will fix but realized later that it isn't correct and didnt wat to delete commit by force pushing

**Pull request closed without merging**